cw_wownero/ios/Classes/wownero_api.cpp

```diff
@@ -0,0 +1,1034 @@
+    {
+        m_last_known_wallet_height = height;
+    }
+
+    return height;
+}
+
+uint64_t is_needed_to_refresh()
+{
+    if (m_listener == nullptr) {
+        return false;
+    }
+
+    bool should_refresh = m_listener->isNeedToRefresh();
+
+    if (should_refresh) {
+        m_listener->resetNeedToRefresh();
+    }
+
+    return should_refresh;
+}
+
+uint8_t is_new_transaction_exist()
+{
+    if (m_listener == nullptr) {
+        return false;
+    }
+
+    bool is_new_transaction_exist = m_listener->isNewTransactionExist();
+
+    if (is_new_transaction_exist)
+    {
+        m_listener->resetIsNewTransactionExist();
+    }
+
+    return is_new_transaction_exist;
+}
+
+void set_listener()
+{
+    m_last_known_wallet_height = 0;
+
+    if (m_listener != nullptr)
+    {
+        free(m_listener);
+    }
+
+    m_listener = new MoneroWalletListener();
+    get_current_wallet()->setListener(m_listener);
+}
+
+int64_t *subaddrress_get_all()
+{
+    std::vector<Monero::SubaddressRow *> _subaddresses = m_subaddress->getAll();
+    size_t size = _subaddresses.size();
+    int64_t *subaddresses = (int64_t *)malloc(size * sizeof(int64_t));
+
+    for (int i = 0; i < size; i++)
+    {
+        Monero::SubaddressRow *row = _subaddresses[i];
+        SubaddressRow *_row = new SubaddressRow(row->getRowId(), strdup(row->getAddress().c_str()), strdup(row->getLabel().c_str()));
+        subaddresses[i] = reinterpret_cast<int64_t>(_row);
+    }
+
+    return subaddresses;
+}
+
+int32_t subaddrress_size()
+{
+    std::vector<Monero::SubaddressRow *> _subaddresses = m_subaddress->getAll();
+    return _subaddresses.size();
+}
+
+void subaddress_add_row(uint32_t accountIndex, char *label)
+{
+    m_subaddress->addRow(accountIndex, std::string(label));
+}
+
+void subaddress_set_label(uint32_t accountIndex, uint32_t addressIndex, char *label)
+{
+    m_subaddress->setLabel(accountIndex, addressIndex, std::string(label));
+}
+
+void subaddress_refresh(uint32_t accountIndex)
+{
+    m_subaddress->refresh(accountIndex);
+}
+
+int32_t account_size()
+{
+    std::vector<Monero::SubaddressAccountRow *> _accocunts = m_account->getAll();
+    return _accocunts.size();
+}
+
+int64_t *account_get_all()
+{
+    std::vector<Monero::SubaddressAccountRow *> _accocunts = m_account->getAll();
+    size_t size = _accocunts.size();
+    int64_t *accocunts = (int64_t *)malloc(size * sizeof(int64_t));
+
+    for (int i = 0; i < size; i++)
+    {
+        Monero::SubaddressAccountRow *row = _accocunts[i];
+        AccountRow *_row = new AccountRow(row->getRowId(), strdup(row->getLabel().c_str()));
+        accocunts[i] = reinterpret_cast<int64_t>(_row);
+    }
+
+    return accocunts;
+}
+
+void account_add_row(char *label)
+{
+    m_account->addRow(std::string(label));
+}
+
+void account_set_label_row(uint32_t account_index, char *label)
+{
+    m_account->setLabel(account_index, label);
+}
+
+void account_refresh()
+{
+    m_account->refresh();
```

Comment:
@OmarHatem28 
idk where to remove accounts. Can you guys handle

1. Removing accounts
2. Adding to fiat api
3. Adding to exchange